packages/flutter/lib/src/scheduler/binding.dart

```diff
@@ -954,20 +954,39 @@ mixin SchedulerBinding on BindingBase {
 
   int _debugFrameNumber = 0;
   String? _debugBanner;
-  bool _ignoreNextEngineDrawFrame = false;
+
+  // Whether the current engine frame needs to be postponed till after the
+  // warm-up frame.
+  //
+  // Engine may begin a frame in the middle of the warm-up frame because the
+  // warm-up frame is scheduled by timers while the engine frame is scheduled
+  // by platform specific frame scheduler (e.g. `requestAnimationFrame` on the
+  // web). When this happens, we let the warm-up frame finish, and postpone the
+  // engine frame.
+  bool _rescheduleAfterWarmUpFrame = false;
 
   void _handleBeginFrame(Duration rawTimeStamp) {
     if (_warmUpFrame) {
-      assert(!_ignoreNextEngineDrawFrame);
-      _ignoreNextEngineDrawFrame = true;
+      // "begin frame" and "draw frame" must strictly alternate. Therefore
+      // _rescheduleAfterWarmUpFrame cannot possibly be true here as it is
+      // reset by _handleDrawFrame.
+      assert(!_rescheduleAfterWarmUpFrame);
+      _rescheduleAfterWarmUpFrame = true;
       return;
     }
     handleBeginFrame(rawTimeStamp);
   }
 
   void _handleDrawFrame() {
-    if (_ignoreNextEngineDrawFrame) {
-      _ignoreNextEngineDrawFrame = false;
+    if (_rescheduleAfterWarmUpFrame) {
+      _rescheduleAfterWarmUpFrame = false;
+      // Reschedule in a timer to allow the draw-frame phase of the warm-up
+      // frame to finish.
+      Timer.run(() {
+        // Force a new frame no matter what state _hasScheduledFrame is in.
```

Comment:
Why do we need to schedule another frame if one is already scheduled?

Reply:
(In other words, why do we ignore _hasScheduledFrame here?)

Reply:
From GVC: We should say that we need to reset it here because it was never reseted in _handleBeginFrame when we dropped the frame.

Reply:
Actually, after discussing this a little more, this is fine. We have no idea what state _hasScheduledFrame is in, and we need it to be false so that scheduleFrame actually does something.

Reply:
Added more explanation.